jvm/core/src/main/java/ml/dmlc/tvm/contrib/GraphRuntime.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ml.dmlc.tvm.contrib;
+
+import ml.dmlc.tvm.Function;
+import ml.dmlc.tvm.Module;
+import ml.dmlc.tvm.TVMContext;
+import ml.dmlc.tvm.TVMValue;
+import ml.dmlc.tvm.rpc.RPC;
+import ml.dmlc.tvm.rpc.RPCSession;
+import ml.dmlc.tvm.rpc.TVMRemoteContext;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+public class GraphRuntime {
+  /**
+   * Create a runtime executor module given a graph and module.
+   * @param graphJson The graph deployed in json format output by nnvm graph.
+   * @param libmod The module of the corresponding function.
+   * @param ctx The local or remote context to deploy the module.
+   * @return Runtime graph module that can be used to execute the graph.
+   */
+  public static GraphModule create(String graphJson, Module libmod, TVMContext ctx) {
+    Module graphModule = null;
+    if (ctx.deviceType >= RPC.RPC_SESS_MASK) {
```

Comment:
Maybe combine these into `if P and Q` instead of two separate `if` statements?

Reply:
I feel the current statement is more readable, the `if`s inside are more about `assert`, rather than control flows.